src/python_ml_project_template/datasets/flow_trajectory_dataset_pyg.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Optional, Protocol, Union, cast
+
+import rpad.partnet_mobility_utils.dataset as pmd
+import torch
+import torch_geometric.data as tgd
+
+from python_ml_project_template.datasets.flow_trajectory_dataset import (
+    FlowTrajectoryDataset,
+)
+
+
+class FlowTrajectoryTGData(Protocol):
+    id: str  # Object ID.
+
+    pos: torch.Tensor  # Points in the point cloud.
+    trajectory: torch.Tensor  # instantaneous positive 3D flow trajectories.
+    mask: torch.Tensor  # Mask of the part of interest.
+
+
+class FlowTrajectoryPyGDataset(tgd.Dataset):
+    def __init__(
+        self,
+        root: str,
+        split: Union[pmd.AVAILABLE_DATASET, List[str]],
+        randomize_joints: bool = True,
+        randomize_camera: bool = True,
+        trajectory_len: int = 5,
+        mode: str = "delta",
+        n_points: Optional[int] = 1200,
+    ) -> None:
+        super().__init__()
+        self.dataset = FlowTrajectoryDataset(
+            root,
+            split,
+            randomize_joints,
+            randomize_camera,
+            trajectory_len,
+            mode,
+            n_points,
+        )
+
+    def len(self) -> int:
+        return len(self.dataset)
+
+    def get(self, index) -> tgd.Data:
+        return self.get_data(self.dataset._dataset._ids[index], seed=None)
+
+    @staticmethod
+    def get_processed_dir(randomize_joints, randomize_camera):
+        joint_chunk = "rj" if randomize_joints else "sj"
```

Comment:
you should change this directory, otherwise it's gonna load from the original one. so make a change to the final fstring, call it "processed_traj" or something